AsyncDisplayKit/ASCellNode.m

```diff
@@ -32,6 +40,35 @@ - (instancetype)init
   return self;
 }
 
+- (instancetype)initWithViewControllerBlock:(ASDisplayNodeViewControllerBlock)viewControllerBlock
+{
+  return [self initWithViewControllerBlock:viewControllerBlock didLoadBlock:nil];
+}
+
+- (instancetype)initWithViewControllerBlock:(ASDisplayNodeViewControllerBlock)viewControllerBlock didLoadBlock:(ASDisplayNodeDidLoadBlock)didLoadBlock
+{
+  if (!(self = [super init]))
+    return nil;
+  
+  ASDisplayNodeAssertNotNil(viewControllerBlock, @"should initialize with a valid block that returns a UIViewController");
+  
+  _viewController = viewControllerBlock();
+  
+  _viewControllerNode = [[ASDisplayNode alloc] initWithViewBlock:^UIView *{
+        return _viewController.view;
+      } didLoadBlock:didLoadBlock];
+  
+  [self addSubnode:_viewControllerNode];
+  
+  _nodeLoadedBlock = didLoadBlock;
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize {
```

Comment:
Brace on next line :)

It is very important not to set a frame in this method.  This method is called off the main thread, and as written today, you can't setFrame: on a node after it is loaded (as it sends it to UIView after the UIView exists).

You should instead not actually implement layoutSpecThatFits:, but instead implement -layout.  Call super in the method, then you can set _viewControllerNode.frame = self.bounds;  Fortunately -layout is always guaranteed to be called on the main thread.

This seems to me like the best implementation as it could be possible for a subclass to implement an actual -layoutSpecThatFits, and it's a bit tough to add the viewControllerNode to an existing spec.  Honestly though it could be done using ASOverlayLayoutSpec, but it doesn't seem necessary.